backend/rfc8291Ikm.js

```diff
@@ -0,0 +1,80 @@
+// Key derivation as per RFC 8291 (for sending encrypted push notifications)
+export default async (uaPublic: Uint8Array, salt: SharedArrayBuffer): Promise<[ArrayBuffer, ArrayBuffer]> => {
```

Comment:
Claude:

> The `salt` parameter is typed as `SharedArrayBuffer` but in the usage from `push.js` it's being passed a regular `Buffer`. This could cause type checking issues and potential runtime problems. The type should probably be:
> 
> ```js
> export default async (uaPublic: Uint8Array, salt: Buffer | Uint8Array): Promise<[ArrayBuffer, ArrayBuffer]> => {
> ```